src/main/java/com/onarandombox/multiverseinventories/util/CommentedYamlConfiguration.java

```diff
@@ -160,116 +155,85 @@ public boolean save() {
-                     * May need reworking.
-                    if (comment != null || (line.length() > 1 && line.charAt(line.length() - 1) == ':')) {
-                        // Add the (modified) line to the total config String
-                        // This modified version will not write the config if a comment is not present
-                        newContents += line + System.getProperty("line.separator");
+                        newLine.insert(0, "\n").insert(0, comment);
+                        if (newLine.charAt(newLine.length() - 1) != ':' && !line.equals(arrayConfig[arrayConfig.length - 1]))
+                            newLine.append("\n");
                     }
-                    */
                 }
-                newLine.append(System.getProperty("line.separator"));
+
+                newLine.append("\n");
                 // Add the (modified) line to the total config String
                 newContents.append(newLine.toString());
             }
-            /*
-             * Due to a bukkit bug we need to strip any extra new lines from the
-             * beginning of this file, else they will multiply.
-             */
-            /*
-            while (newContents.startsWith(System.getProperty("line.separator"))) {
-                newContents = newContents.replaceFirst(System.getProperty("line.separator"), "");
-            }
-            */
-            try {
-                // Write the string to the config file
-                this.stringToFile(newContents.toString(), file);
-            } catch (IOException e) {
-                saved = false;
-            }
+
+            saved = this.stringToFile(newContents.toString(), file);
         }
+
         return saved;
     }
 
     /**
-     * Adds a comment just before the specified path.  The comment can be multiple lines.  An empty string will indicate a blank line.
+     * Adds a comment just before the specified path. The comment can be multiple lines. An empty string will indicate
+     * a blank line.
      *
      * @param path         Configuration path to add comment.
-     * @param commentLines Comments to add.  One String per line.
+     * @param commentLines Comments to add. One String per line.
      */
     public void addComment(String path, List<String> commentLines) {
-        StringBuilder commentstring = new StringBuilder();
-        String leadingSpaces = "";
+        StringBuilder commentString = new StringBuilder();
+        StringBuilder leadingSpaces = new StringBuilder();
         for (int n = 0; n < path.length(); n++) {
             if (path.charAt(n) == '.') {
-                leadingSpaces += "  ";
+                leadingSpaces.append("  ");
             }
         }
         for (String line : commentLines) {
             if (!line.isEmpty()) {
-                line = leadingSpaces + line;
-            } else {
-                line = " ";
+                line = leadingSpaces.toString() + line;
```

Comment:
Use the StringBuilder of `commentString` here too?

Reply:
Yeah I ended up doing it a bit differently, but same idea.

Reply:
Actually never mind, I get you meant.